__tests__/app.test.js

```diff
@@ -51,4 +53,39 @@ describe('GET: /api', ()=> {
         })
     })
 })
+describe('GET /api/articles/:article_id', () => {
+    test('GET 200: returns a single article object', () => {
+        return request(app)
+        .get('/api/articles/4')
+        .expect(200)
+        .then(({body}) => {
+            const article = body
+            expect(typeof article).toBe('object')
+            expect(Array.isArray(article)).toBe(false)
+        })
+    })
+    test('GET 200: returns a single article object, corresponding to the correct id with all the correct properties and data types', () => {
+        return request(app)
+        .get('/api/articles/4')
+        .expect(200)
+        .then(({body}) => {
+            const article = body
+            expect(article.article_id).toEqual(4)
+            expect(typeof article.author).toBe('string')
+            expect(typeof article.title).toBe('string')
+            expect(typeof article.body).toBe('string')
+            expect(typeof article.topic).toBe('string')
+            expect(typeof article.created_at).toBe('string')
+            expect(typeof article.votes).toBe('number')
+            expect(typeof article.article_img_url).toBe('string')
+        })
+    })
+    test('GET 404: Returns a not found error when passed a valid but non-existent id', () => {
+        return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('not found')
+        })
+    })
```

Comment:
Make sure you add in a test for a 400: Bad request. For example - **E.g. GET /api/articles/notanumber**